pkg/policyviolation/clusterpv.go

```diff
@@ -271,20 +274,20 @@ func containsOwner(owners []kyverno.ResourceSpec, pv *kyverno.ClusterPolicyViola
 
 // validDependantForDeployment checks if resource (pod) matches the intent of the given deployment
 // explicitly handles deployment-replicaset-pod relationship
-func validDependantForDeployment(client appsv1.AppsV1Interface, curPv kyverno.ClusterPolicyViolation, resource unstructured.Unstructured) bool {
+func validDependantForDeployment(client appsv1.AppsV1Interface, pvResourceSpec kyverno.ResourceSpec, resource unstructured.Unstructured) bool {
 	if resource.GetKind() != "Pod" {
 		return false
 	}
 
 	// only handles deploymeny-replicaset-pod relationship
-	if curPv.Spec.ResourceSpec.Kind != "Deployment" {
+	if pvResourceSpec.Kind != "Deployment" {
 		return false
 	}
 
 	owner := kyverno.ResourceSpec{
-		Kind:      curPv.Spec.ResourceSpec.Kind,
-		Namespace: curPv.Spec.ResourceSpec.Namespace,
-		Name:      curPv.Spec.ResourceSpec.Name,
+		Kind:      pvResourceSpec.Kind,
+		Namespace: pvResourceSpec.Namespace,
+		Name:      pvResourceSpec.Name,
 	}
 
 	deploy, err := client.Deployments(owner.Namespace).Get(owner.Name, metav1.GetOptions{})
```

Comment:
As we know the types, can use shared informer on deployment resource or if this is intentional then dynamic client

Reply:
This client is used in external library to get current valid replicaset for the given deploy, add TODO to remove.
https://github.com/nirmata/kyverno/blob/e2af3852f950de5e8b4a6fc8a7739c0324ab1355/pkg/policyviolation/helpers.go#L96-L101

Reply:
I was initially suggesting to use a shared informer on the resource as it will work on the local cache and faster. But using a typed client is ok.